pkg/apiserver/handlers.go

```diff
@@ -430,13 +430,18 @@ func WithActingAs(handler http.Handler, requestContextMapper api.RequestContextM
 		}
 
 		actingAsAttributes := &authorizer.AttributesRecord{
-			User:     requestor,
-			Verb:     "impersonate",
-			APIGroup: api.GroupName,
-			Resource: "users",
-			// ResourceName:    requestedSubject,
+			User:            requestor,
+			Verb:            "impersonate",
+			APIGroup:        api.GroupName,
+			Resource:        "users",
+			Name:            requestedSubject,
 			ResourceRequest: true,
 		}
+		if namespace, name, err := serviceaccount.SplitUsername(requestedSubject); err == nil {
+			actingAsAttributes.Resource = "serviceaccounts"
+			actingAsAttributes.Namespace = namespace
+			actingAsAttributes.Name = name
+		}
```

Comment:
I have a slight preference for an if/else rather than setting to `users/<name>` and overriding here, but otherwise LGTM